From 2e77e04dfeedc8c1b2c68389c228a169eacc1ecb Mon Sep 17 00:00:00 2001
From: Jaroslav Kysela <jkysela@redhat.com>
Date: Wed, 3 Sep 2014 07:30:11 -0400
Subject: [alsa] usb-audio: Suppress repetitive debug messages from retire_playback_urb()

Message-id: <140972823573-265-patchset-jkysela@redhat.com>
Patchwork-id: 90967
O-Subject: [RHEL 7.1 PATCH 264/442] ALSA: usb-audio: Suppress repetitive debug messages from retire_playback_urb()
Bugzilla: 1112200
RH-Acked-by: John Linville <linville@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Ivan Vecera <ivecera@redhat.com>

BugLink: http://bugs.launchpad.net/bugs/1305133

Malfunctioning or slow devices can cause a flood of dmesg SPAM.

I've ignored checkpatch.pl complaints about the use of printk_ratelimit() in favour
of prior art in sound/usb/pcm.c.

WARNING: Prefer printk_ratelimited or pr_<level>_ratelimited to printk_ratelimit
+ if (printk_ratelimit() &&

Cc: Jaroslav Kysela <perex@perex.cz>
Cc: Takashi Iwai <tiwai@suse.de>
Cc: Eldad Zack <eldad@fogrefinery.com>
Cc: Daniel Mack <zonque@gmail.com>
Cc: Clemens Ladisch <clemens@ladisch.de>
Signed-off-by: Tim Gardner <tim.gardner@canonical.com>
Signed-off-by: Takashi Iwai <tiwai@suse.de>

(picked from kernel: a5065eb6da55b226661456e6a7435f605df98111)
rhbz#1112200
Signed-off-by: Jaroslav Kysela <jkysela@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/sound/usb/pcm.c b/sound/usb/pcm.c
index 49de5c1..131336d 100644
--- a/sound/usb/pcm.c
+++ b/sound/usb/pcm.c
@@ -1501,7 +1501,8 @@ static void retire_playback_urb(struct snd_usb_substream *subs,
   * The error should be lower than 2ms since the estimate relies
   * on two reads of a counter updated every ms.
   */
- if (abs(est_delay - subs->last_delay) * 1000 > runtime->rate * 2)
+ if (printk_ratelimit() &&
+     abs(est_delay - subs->last_delay) * 1000 > runtime->rate * 2)
   dev_dbg(&subs->dev->dev,
    "delay: estimated %d, actual %d\n",
    est_delay, subs->last_delay);
-- 
1.7.1